server/pruning.go

```diff
@@ -36,3 +43,71 @@ func GetPruningOptionsFromFlags(appOpts types.AppOptions) (pruningtypes.PruningO
 		return pruningtypes.PruningOptions{}, fmt.Errorf("unknown pruning strategy %s", strategy)
 	}
 }
+
+// PruningCmd prunes the sdk root multi store history versions based on the pruning options
+// specified by command flags.
+func PruningCmd(providerCreator types.StoreProviderCreator) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "prune",
+		Short: "prune history stetes based on the pruning options specified by flags",
+		Long: `Pruning options can be provided via the '--pruning' flag or alternatively with '--pruning-keep-recent', and
+		'pruning-interval' together.
+		
+		For '--pruning' the options are as follows:
+		
+		default: the last 362880 states are kept, pruning at 10 block intervals
+		nothing: all historic states will be saved, nothing will be deleted (i.e. archiving node)
+		everything: 2 latest states will be kept; pruning at 10 block intervals.
+		custom: allow pruning options to be manually specified through 'pruning-keep-recent', and 'pruning-interval'
+		`,
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			serverCtx := GetServerContextFromCmd(cmd)
+
+			// Bind flags to the Context's Viper so the app construction can set
+			// options accordingly.
+			if err := serverCtx.Viper.BindPFlags(cmd.Flags()); err != nil {
+				return err
+			}
+
+			_, err := GetPruningOptionsFromFlags(serverCtx.Viper)
+			return err
+		},
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			ctx := GetServerContextFromCmd(cmd)
+
+			home := ctx.Viper.GetString(flags.FlagHome)
+			db, err := openDB(home, GetAppDBBackend(ctx.Viper))
+			if err != nil {
+				return err
+			}
+
+			// we should set pruning options in providerCreator
+			provider := providerCreator(ctx.Logger, db, ctx.Viper)
+			cms := provider.CommitMultiStore()
+			cmsOptions := cms.GetPruning()
+			// set pruning options for cms in case we forgot to apply the pruning options in providerCreator
+			if cmsOptions.Strategy == 0 && cmsOptions.KeepRecent == 0 {
+				pruningOptions, err := GetPruningOptionsFromFlags(ctx.Viper)
+				if err != nil {
+					return err
+				}
+				cms.SetPruning(pruningOptions)
+			}
+
+			if rootMultiStore, ok := cms.(*rootmulti.Store); ok {
+				err = rootMultiStore.PruneHistoryVersions()
+				return err
+			}
+
+			return fmt.Errorf("currently only support the pruning of rootmulti.Store type")
+		},
+	}
+
+	cmd.Flags().String(flags.FlagHome, "", "The database home directory")
+	cmd.Flags().String(FlagPruning, pruningtypes.PruningOptionDefault, "Pruning strategy (default|nothing|everything|custom)")
+	cmd.Flags().Uint64(FlagPruningKeepRecent, 0, "Number of recent heights to keep on disk (ignored if pruning is not 'custom')")
+	cmd.Flags().Uint64(FlagPruningInterval, 0, "Height interval at which pruned heights are removed from disk (ignored if pruning is not 'custom')")
```

Comment:
is pruning interval needed? I would assume we would want to prune right away so we should default to right away here

Reply:
maybe we can quickly replay the block by keeping some intermediate states instead of pruning all the lower version states?

Reply:
I think we should read the config items from`app.toml`, to keep the running result consistent with the pruning config.

Reply:
> is pruning interval needed? I would assume we would want to prune right away so we should default to right away here

This is just for reading the config items from the `app.toml`, will prune according to whatever strategy configured by user.